pydevices/HtsDevices/cryocon18i.py

```diff
@@ -241,30 +236,29 @@ def store(self):
 
         # Saving TREND shot number information into the tree:
         self.trend_shot.record = trend_tree.getCurrent(self.trend_tree.data())
-        
+
         # Set Time Context
         trend_tree.setTimeContext(t1, MDSplus.Int64(self.t2.data()*1000.))
 
         print('Writing data into shot node')
-        for c in self.inputs():
-            trend_temp     = trend_dev.__getattr__('input_{}'.format(c))
-            trend_resis    = trend_dev.__getattr__('input_{}_resistence'.format(c))
-                        
+        for i in self.inputs:
+            trend_temp     = trend_dev.__getattr__('input_%c'.format(c))
+            trend_resis    = trend_dev.__getattr__('input_%c_resistence'.format(c))
+
             times    = trend_temp.dim_of().data()
             temps    = trend_temp.data()
             resists  = trend_resis.data()
 
             start_time = times[0]
             for j in range(len(times)):
                 times[j] -= start_time
-                times[j] = float(times[j]) / 1000.    
+                times[j] = float(times[j]) / 1000.
+
+            shot_temp     = self.__getattr__('input_%c' % (i,))
```

Comment:
Why use format above but not here ?  I don't understand the  tradeoffs between various format syntaxes

Reply:
never use .format

Reply:
real issue was the c instead of i